test/methods/gd_non_sequential_armijo_safe_bb.jl

```diff
@@ -0,0 +1,1691 @@
+                )
+
+            for (field_key, field_type) in field_info(type)
+                if field_key != :bb_step_size
+                    @test field_type == typeof(getfield(optData, field_key))
+                end
+            end
+            
+            # random field values -- long_stepsize = false
+            x0 = randn(type, 10)
+            long_stepsize = false
+            α_lower = abs(rand(type))
+            α_default = type(1.0)
+            init_stepsize = type((α_lower + 1/α_lower)/2)
+            δ0 = abs(rand(type))
+            δ_upper = δ0 + 1
+            ρ = abs(rand(type))
+            M = rand(1:100)
+            threshold = abs(rand(type))
+            max_iterations =  rand(1:100)
+    
+            optData = NonsequentialArmijoSafeBBGD(type; 
+                x0 = x0,
+                init_stepsize = init_stepsize, 
+                long_stepsize = long_stepsize, 
+                α_lower = α_lower,
+                α_default = α_default,
+                δ0 = δ0,
+                δ_upper = δ_upper,
+                ρ = ρ, 
+                M = M,
+                threshold = threshold,
+                max_iterations = max_iterations
+                )
+            
+            for (field_key, field_type) in field_info(type)
+                if field_key != :bb_step_size
+                    @test field_type == typeof(getfield(optData, field_key))
+                end
+            end
+        end
+    end
+
+    # test struct initializations
+    let
+
+        # random field values -- long_stepsize = true
+        x0 = randn(10)
+        long_stepsize = true
+        α_lower = abs(rand())
+        α_default = 1.0
+        init_stepsize = (α_lower + 1/α_lower)/2
+        δ0 = abs(rand())
+        δ_upper = δ0 + 1
+        ρ = rand()
+        M = rand(1:100)
+        threshold = abs(rand())
+        max_iterations = rand(1:100) 
+
+        optData = NonsequentialArmijoSafeBBGD(Float64; 
+            x0 = x0,
+            init_stepsize = init_stepsize, 
+            long_stepsize = long_stepsize, 
+            α_lower = α_lower,
+            α_default = α_default,
+            δ0 = δ0,
+            δ_upper = δ_upper,
+            ρ = ρ, 
+            M = M,
+            threshold = threshold,
+            max_iterations = max_iterations
+            )
+
+        # test field value correct initialized long_stepsize = true
+        @test optData.iter_hist[1] == x0
+        @test optData.init_stepsize == init_stepsize
+        @test optData.bb_step_size == OptimizationMethods.bb_long_step_size
+        @test optData.α_lower == α_lower
+        @test optData.α_default == α_default
+        @test optData.δk == δ0
+        @test optData.δ_upper == δ_upper
+        @test optData.ρ == ρ
+        @test length(optData.objective_hist) == M
+        @test optData.threshold == threshold
+        @test optData.max_iterations == max_iterations
+        
+        # random field values -- long_stepsize = false
```

Comment:
Should be in its own `let` statement with the arguments as comma separated arguments for the `let` block